src/coreclr/vm/jithelpers.cpp

```diff
@@ -5375,7 +5375,7 @@ NOINLINE static void JIT_ReversePInvokeEnterRare(ReversePInvokeFrame* frame, voi
     frame->currentThread = thread;
 
 #ifdef PROFILING_SUPPORTED
-        if (CORProfilerTrackTransitions())
+        if (frame->pMD && CORProfilerTrackTransitions())
```

Comment:
Do we still need the `frame->pMD` null check here? It is not in other similar places. I am not able to figure what makes this place special.

Reply:
This function is used from both of the Enter helpers. I’m using a null pMD as the semantic for being called from the non-transition-tracking helper.

Reply:
I think that MetthodDesc is null for the transition tracking helper in the UnmanagedCallersOnly case. It should be sufficient to check `CORProfilerTrackTransitions` here. The value returned by `CORProfilerTrackTransitions` cannot change during the process lifetime.